### PR TITLE
Added Teacher model, repo, service, and controller

### PR DESCRIPTION
**Pull request by LiveNathan**

### CODE REVIEW

File: src/main/java/cnLabs/reactivedemo/Controller/TeacherController.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import static org.springframework.web.reactive.function.server.RequestPredicates.*;
+import static org.springframework.web.reactive.function.server.RouterFunctions.route;
+import static org.springframework.web.reactive.function.server.ServerResponse.ok;
+
+@Component
+public class TeacherController {
+
+    @Autowired
+    TeacherService teacherService;
```
Comment (NikitaBaranov) [inline src/main/java/cnLabs/reactivedemo/Controller/TeacherController.java]:
```suggestion
    private final TeacherService teacherService;
```

File: src/main/java/cnLabs/reactivedemo/Controller/TeacherController.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        return route(GET("/teachers/{id}"),
+                req -> ok()
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .body(teacherService.getTeacher(Integer.parseInt(req.pathVariable("id"))), Teacher.class));
+    }
+
+    @Bean
+    public RouterFunction<ServerResponse> addNewTeacher() {
+        return route(POST("/teachers"), req ->
+                req.bodyToMono(Teacher.class)
```
Comment (NikitaBaranov) [inline src/main/java/cnLabs/reactivedemo/Controller/TeacherController.java]: Try to be consistent with Code formatting. 
The more consistent you are with code, the easier to maintain

```suggestion
        return route(POST("/teachers"),
                 req -> req.bodyToMono(Teacher.class)
```

**Review by NikitaBaranov — APPROVED** [review]
All good! As usual :) Left some formatting comments

File: src/main/java/cnLabs/reactivedemo/Controller/TeacherController.java (hunk 1 above)
Comment (LiveNathan) [reply src/main/java/cnLabs/reactivedemo/Controller/TeacherController.java]: Oh, should the services autowired into Controllers always be final?

**Pull request merged**